libminifi/include/utils/ClassUtils.h

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef LIBMINIFI_INCLUDE_UTILS_CLASSUTILS_H_
+#define LIBMINIFI_INCLUDE_UTILS_CLASSUTILS_H_
+
+#include <string>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+namespace ClassUtils {
+
+/**
+ * Shortens class names via the canonical representation
+ * @param class_name input class name
+ * @param out output class name that is shortened.
+ */
+extern void shortenClassName(const std::string &class_name, std::string &out);
```

Comment:
Why do we need extern here? Libminifi build should be wholearchive anyway.

Reply:
Thanks for pointing this out! It's not needed as is superfluous in header files.

 It was copied from the original location ( the source file ), in which case the use of wholarchive didn't help locate during the linking phase -- but when rebasing this PR I moved this to the header, at which point it's actually already externalized across translation units.